api-lib/handleOptOutMsg.ts

```diff
@@ -1,7 +1,57 @@
+import { isFeatureEnabled } from '../src/config/features';
+
 import * as queries from './gql/queries';
-import { sendSocialMessage } from './sendSocialMessage';
+import {
+  Channels,
+  DiscordOptsOut,
+  sendSocialMessage,
+} from './sendSocialMessage';
+import { Awaited } from './ts4.5shim';
 import { EventTriggerPayload } from './types';
 
+type GetChannelsProps = {
+  data: EventTriggerPayload<'users', 'UPDATE'>['event']['data'];
+  circle: Awaited<ReturnType<typeof queries.getCircle>>['circles_by_pk'];
+  profiles: Awaited<
+    ReturnType<typeof queries.getProfileAndMembership>
+  >['profiles'];
+  channels: Channels<DiscordOptsOut>;
+};
+
+function getChannels(props: GetChannelsProps): Channels<DiscordOptsOut> {
+  const { channels, circle, data, profiles } = props || {};
+
+  if (isFeatureEnabled('discord') && channels?.discordBot) {
+    const { discord_channel_id: channelId, discord_role_id: roleId } =
+      circle?.discord_circle || {};
+
+    const user = profiles[0].user;
+
+    if (!channelId || !roleId || !user) {
+      return null;
+    }
+
+    return {
+      discordBot: {
+        type: 'user-opts-out' as const,
+        channelId,
+        roleId,
+        discordId: user.user_snowflake,
+        address: data.new.address,
+        circleName: circle?.name ?? 'Unknown',
+        // TODO Where to get this?
+        refunds: [
+          { username: 'Alice', give: 10 },
+          { username: 'Bob', give: 15 },
+          { username: 'Mallory', give: 75 },
+        ],
```

Comment:
@topocount do you know how to get this refund data?

Reply:
@karelianpie unfortunately we don't store it anywhere easily accessible. We just emit a message using this event trigger and get the values from the event trigger payload: https://github.com/coordinape/coordinape/blob/epoch-cron/api-lib/handleRefundGiveMsg.ts

Reply:
This means that each refund quantity is a separate message from the bot

Reply:
The voucher name, or nomination reason for example. I think those are required, so we should have them AFAIK.

Reply:
@reeserj would like to have the refunds, so we can just have that data in the handleOptOutMsg? You've mentioned @topocount that would be easy and you could do this while adding the user added/removed from circle

Reply:
Refunds done with https://github.com/coordinape/coordinape/pull/1838/commits/4b59ed3ff78127e4cf7b5ebd33188d4e867365a6 and https://github.com/coordinape/coordinape/pull/1838/commits/850d307c08b862cc7c03f04ee700e10fa53101a9